File: python/crypto.py
import string

def string_value(x):
    alphabet = string.ascii_lowercase
    return sum([alphabet.index(i)+1 for i in x.lower()])
    
def rot_shift(text, n):
    alphabet = string.ascii_lowercase
    shifted_alphabet = alphabet[n:] + alphabet[:n]
    table = str.maketrans(alphabet, shifted_alphabet)
    return text.translate(table)

def perm(n, key):
    num = ""
    m = 1
    while True:
        num = str(n%m) + num
        n=(n//m)
        m+=1
        if n==0:
            break
    tup2 = [0]*len(key)
    tup2[0] = key[int(num[0])]
    key.pop(int(num[0]))
    for i,j in enumerate(num[1:]):
        tup2[i+1] = key[int(j)]
        key.pop(int(j))
    return tup2

File: java/Crypto.java

public class Crypto {
    public static int str_value(String x) {
        int sum = 0;
        for (char y : x.toLowerCase().toCharArray()) {
            sum += (int) (y) - 96;
        }
        return sum;
    }

    public static int str_value(char x) {
        return (int) (x) - 96;
    }

    public static String rot_shift(String text, int n) {
        String string2 = "";
        for (char y : text.toCharArray()) {
            string2 += (char) ((int) (y) + n);
        }
        return string2;
    }

    public static String rot_shift_alpha(String text, int n) {
        String string2 = "";
        for (char y : text.toLowerCase().toCharArray()) {
            int z = (int) (y) + n;
            if (z < 97) {
                string2 += (char) (123 - (97 - z));
            } else if (z > 122) {
                string2 += (char) (97 + (123 - z));
            } else {
                string2 += (char) z;
            }
        }
        return string2;
    }

}
